`ValueError` for `setup.cfg` with both explicit `[options] packages = ...` and unused `[options.packages.find]`

Some packages have `setup.cfg` with explicit `[options] packages = ...` but unused `[options.packages.find]`.

In general, `setup.cfg` should either list packages explicitly or use `find`,
but sometimes configs change and `options.packages.find` is left behind.

This can be seen for example in `aiohttp`:

```ini
# ...
[options]
packages = aiohttp
# ...
[options.packages.find]
exclude =
  examples
# ...
```

Currently `ini2toml` will fail to convert https://github.com/aio-libs/aiohttp/blob/v3.9.5/setup.cfg:

```
cd /tmp
wget https://github.com/aio-libs/aiohttp/raw/v3.9.5/setup.cfg
pipx run 'ini2toml[lite]==0.13' setup.cfg
# [ERROR] ValueError: Value for `packages` expected to be Mapping, found ['aiohttp']
```

A possible solution is for `ini2toml` to remove  `[options.packages.find]` and warn about it.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -2,6 +2,14 @@
 Changelog
 =========
 
+Development Version
+===================
+* ``setuptools`` plugin:
+  * Fix ``ValueError`` when ``setup.cfg`` contains ``[options.packages.find]``
+    but also lists ``[options] packages = ...`` explicitly as a list of package
+    names, :issue:`93`
+
+
 Version 0.13
 ============
 

--- a/src/ini2toml/plugins/setuptools_pep621.py
+++ b/src/ini2toml/plugins/setuptools_pep621.py
@@ -438,8 +438,13 @@ class SetuptoolsPEP621:
         options = doc["options"]
         # Abort when not using find or find_namespace
         packages = options.get("packages")
+
         if not isinstance(packages, Directive):
+            if "options.packages.find" in doc:
+                _ConfusingPackagesConfig.emit()
+                doc.pop("options.packages.find", None)
             return doc
+
         prefix = packages.kind.replace("_", "-")
         # Enhancement #1: Unify find and find_namespaces, using `namespaces` as a flag
         options["packages"] = Directive("find", {"namespaces": "namespace" in prefix})
@@ -845,3 +850,24 @@ def _fuse_lines(line1: Commented[List[str]], line2: Commented[List[str]]):
     keep1, keep2 = values1[:-1], values2[1:]
     shared = values1[-1].strip().strip("\\").strip() + " " + values2[0].strip()
     return Commented(keep1 + [shared] + keep2, line2.comment)
+
+
+class _ConfusingPackagesConfig(UserWarning):
+    _MSG = """Confusing configuration `[options.packages.find]`.
+
+    Original configuration sets both:
+
+    - `[options] packages = ...` as a list of named packages
+    - `[options.packages.find]`
+
+    The confusion comes from the fact that `[options.packages.find]` should be used
+    with `[options] packages = find:` or `[options] packages = find_namespace:`.
+
+    Conversion will ignore `[options.packages.find]`, as it cannot be written in the
+    TOML format when `[options] packages = ...` is already given.
+    """
+    __doc__ = _MSG
+
+    @classmethod
+    def emit(cls, msg=_MSG, stacklevel=1):
+        warnings.warn(msg, category=cls, stacklevel=stacklevel + 1)

--- a/tests/plugins/test_setuptools_pep621.py
+++ b/tests/plugins/test_setuptools_pep621.py
@@ -2,7 +2,12 @@ import pytest
 import tomli
 
 from ini2toml.plugins.profile_independent_tasks import remove_empty_table_headers
-from ini2toml.plugins.setuptools_pep621 import Directive, SetuptoolsPEP621, activate
+from ini2toml.plugins.setuptools_pep621 import (
+    Directive,
+    SetuptoolsPEP621,
+    _ConfusingPackagesConfig,
+    activate,
+)
 from ini2toml.translator import Translator
 
 
@@ -738,3 +743,37 @@ def test_deps_with_line_continuation(plugin, parse, convert):
     print("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
     print(doc)
     assert convert(doc) == expected_convert_directives
+
+
+# ----
+
+# Example taken from https://github.com/aio-libs/aiohttp/
+# In general, you should either list packages explicit or use find,
+# but some configs change and `options.packages.find` is left behind
+example_explicit_packages_with_find = """\
+[options]
+packages = aiohttp
+[options.packages.find]
+exclude =
+  examples
+"""
+
+# In those cases, `ini2toml` should remove  `[options.packages.find]`
+# And warn about it
+expected_explicit_packages_with_find = """\
+[tool.setuptools]
+packages = ["aiohttp"]
+"""
+
+
+def test_explicit_packages_with_find(plugin, parse, convert):
+    doc = parse(example_explicit_packages_with_find)
+    print(doc)
+    doc = plugin.apply_value_processing(doc)
+    print("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
+    with pytest.warns(_ConfusingPackagesConfig, match="[options.packages.find]"):
+        doc = plugin.handle_packages_find(doc)
+    doc.rename("options", ("tool", "setuptools"))
+    print("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
+    print(doc)
+    assert convert(doc) == expected_explicit_packages_with_find

